Sources/LogtoClient/LogtoAuthSession/LogtoAuthSession.swift

```diff
@@ -62,35 +62,30 @@ public class LogtoAuthSession {
                 resources: logtoConfig.resources
             )
 
-            // Create session
-            let session = WebAuthSession(url: authUri, callbackURLScheme: redirectUri.scheme) { [self] in
-                guard let callbackUri = $0 else {
-                    completion(.failure(error: Errors.SignIn(type: .authFailed, innerError: $1)))
-                    return
-                }
+            #if !os(macOS)
+                // Create session
+                let session = LogtoWebViewAuthSession(authUri, redirectUri: redirectUri) { [self] in
+                    guard let callbackUri = $0 else {
+                        completion(.failure(error: Errors.SignIn(type: .authFailed, innerError: nil)))
+                        return
+                    }
 
-                handle(callbackUri: callbackUri)
-            }
-
-            if #available(iOS 13.0, *), let session = session as? ASWebAuthenticationSession {
-                session.presentationContextProvider = self.authContext
-                session.prefersEphemeralWebBrowserSession = true
-            }
+                    handle(callbackUri: callbackUri)
+                }
 
-            DispatchQueue.main.async {
-                session.start()
-            }
+                DispatchQueue.main.async {
+                    session.start()
```

Comment:
It seems that this `session.start()` may not be executed in the same thread as the outer `do...catch` block, will the outer `catch` block catch the exception that occurred in `session.start()`?

Reply:
it wont, the async error will be filled in completion callback